splink/internals/clustering.py

```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+
+from splink.internals.connected_components import solve_connected_components
+from splink.internals.database_api import AcceptableInputTableType, DatabaseAPISubClass
+from splink.internals.input_column import InputColumn
+from splink.internals.splink_dataframe import SplinkDataFrame
+
+
+def cluster_pairwise_predictions_at_threshold(
+    nodes: AcceptableInputTableType,
```

Comment:
This should probably eventually allow the input to also be SplinkDataFrame, but i think that's for a wider PR which allows all public-API functions to accept SplinkDataFrames